### ISSUE TITLE
Guest cart results in TOO_MANY_REDIRECTS error

### ISSUE THREAD

**Issue by kozakhs** [thirtybees/thirtybees#1344]
Hello,

in version 1.3.0 when we click on "Cart" it results in TOO_MANY_REDIRECTS error. After some debugging I found out that this error is caused by ParentOrderController, init() method, on this lines:

// Redirect to the good order process
if (Configuration::get('PS_ORDER_PROCESS_TYPE') == 0 && Dispatcher::getInstance()->getController() != 'order') {
      Tools::redirect('index.php?controller=order');  /////// AFTER COMMENTING THIS LINE OUT IT STOPS REDIRECTING
}

If I disable friendly URLs, everything is OK. 

Could anybody help me?

Thank you!

**Comment by getdatakick** [comment]
I can't reproduce this issue. Do you have any specific configuration in place? What does your url-rewrite settings look like? Do you use pretty urls?

**Comment by kozakhs** [comment]
> I can't reproduce this issue. Do you have any specific configuration in place? What does your url-rewrite settings look like? Do you use pretty urls?

Configuration:
Friendly-URL - enabled
URLs with diacritics - disabled
Cannonical redirect - 302 Moved temporarily
Emit seo fields - yes
Apache multi-views - NO
Apache mod_security - NO

I found out problem is in function Dispatcher::getInstance()->getController()... I have a standart 5-steps order (PS_ORDER_PROCESS_TYPE = 0), but this method returns always orderopc (which is one-page order controller), so it will cycle because this condition will never false:

```
if (Configuration::get('PS_ORDER_PROCESS_TYPE') == 0 && Dispatcher::getInstance()->getController() != 'order') {
Tools::redirect('index.php?controller=order'); /////// AFTER COMMENTING THIS LINE OUT IT STOPS REDIRECTING
}
```

I have also tried to regenerate .htaccess file in administration and delete cache, but with no result.

Thank you!

**Comment by getdatakick** [comment]
Looks to me like a misconfigured rewriting -- instead of redirecting to `order` controller, your store redirects to `orderopc` controller instead, resulting in endless loop. 

Check `Friendly URL` settings in your back office `SEO & URLs`

**Comment by kozakhs** [comment]
My Friendly URLs and SEO & URLs settings seems to be alright. I also regenerated the .htaccess file.
But it is really difficult to reproduce this error, because the cart works in 90% of cases. However I was managed to fix this issue by overriding Dispatcher class, getController() method ... I just forced it to return the 'order' controller instead of 'orderopc' ..

I will keep investigating this problem and if find out something, I will let you know... For this moment I am closing this issue

Thank you very much for your reply
